wolfcrypt/src/sha3.c

```diff
@@ -880,6 +899,17 @@ static int wc_Sha3Final(wc_Sha3* sha3, byte* hash, byte p, byte len)
         return BAD_FUNC_ARG;
     }
 
+#ifdef WOLF_CRYPTO_CB
```

Comment:
Remember that SHA3_Final will invoke InitSha3(sha3) at the end which will clear out the devId.  This is a bug in the other SHA implementations as well and needs to fixed.

Reply:
Final calls `InitSha3` which does not touch devId or heap hint. This is not an issue (says David Garske)

Reply:
Sweet!  I must have gotten confused that it was memsetting the struct.  great to hear.  Thx